hooks/useTokenBalance.tsx

```diff
@@ -0,0 +1,47 @@
+import { TokenListItem } from "@/hooks/useTokenList";
+import { BigNumber } from "@ethersproject/bignumber";
+import { multicall } from "@wagmi/core";
+import useSWR from "swr";
+import { useAccount, useNetwork } from "wagmi";
+
+export const ERC20_ABI = [
+  "function balanceOf(address owner) view returns (uint256 balance)",
+];
+
+const useTokenList = ({ tokenList }: { tokenList: TokenListItem[] }) => {
+  const { chain } = useNetwork();
+  const { address } = useAccount();
+
+  const chainId = chain?.id || 1; // default to mainnet if no chain id
+  const { isLoading, error, data } = useSWR(
+    `userTokenList-${chainId}-${tokenList.length}`,
+    async () => {
+      if (!tokenList || tokenList.length === 0) {
+        return;
+      }
+      let multicallArgs = [];
+      for (const token of tokenList) {
+        multicallArgs.push({
+          abi: ERC20_ABI,
+          functionName: "balanceOf",
+          address: token.address,
+          args: [address],
+        });
+      }
+      const data = await multicall({
```

Comment:
Do you know what the difference is between wagmi's `multicall` and `readContracts`? Seems the latter uses the former under the hood, but with some other stuff around it. Wondering which we should use. Another option would be [`watchMulticall`](https://wagmi.sh/core/actions/watchMulticall) / [`watchReadContracts`](https://wagmi.sh/core/actions/watchReadContracts), not sure how these various approaches compare

Reply:
Under the hood readContracts uses multicall and I think the main difference is that reacContracts adds some error handling functionality. Based on [this](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/actions/contracts/readContracts.ts#L21) comment. So, I think I should probably change this to readContracts.

The watch methods look cool. The only benefit would be if we listen to block changes because we currently handle refetching on chain  change.

Reply:
Cool, yea I don't think we need the watch methods currently. Though `multicall` [also](https://github.com/wagmi-dev/wagmi/blob/2beabc6a587880345f62c2fd305b4c1f1b7339e1/packages/core/src/actions/contracts/multicall.ts#L23-L24) supports the `allowFailure` option, so I'm not sure. 

I just asked about the difference [here](https://github.com/wagmi-dev/wagmi/discussions/1652), tracking in https://github.com/ScopeLift/railgun-shielder/issues/39 so marking this resolved